packages/graph/hash_graph/lib/graph/src/datastore/postgres/mod.rs

```diff
@@ -25,22 +32,126 @@ impl PostgresDatabase {
     }
 }
 
+impl PostgresDatabase {
+    async fn insert_version_id(&self, id: &Identifier) -> Result<(), DatastoreError> {
+        sqlx::query(r#"INSERT INTO ids (version_id, base_id) VALUES ($1, $2);"#)
+            .bind(&id.version_id)
+            .bind(&id.base_id)
+            .execute(&self.pool)
+            .await
+            .report()
+            .change_context(DatastoreError)
+            .attach_printable_lazy(|| id.clone())
+            .attach_printable("Could not insert id")?;
+
+        Ok(())
+    }
+
+    async fn insert_data_type(
+        &self,
+        id: &Identifier,
+        data_type: &DataType,
+        created_by: AccountId,
+    ) -> Result<DataTypeRow, DatastoreError> {
+        sqlx::query_as(
+            r#"
+            INSERT INTO data_types (
+                version_id,
+                schema,
+                created_by
+            ) 
+            VALUES ($1, $2, $3)
+            RETURNING version_id, schema, created_by;
+            "#,
+        )
+        .bind(&id.version_id)
+        .bind(&data_type)
+        .bind(&created_by)
+        .fetch_one(&self.pool)
+        .await
+        .report()
+        .change_context(DatastoreError)
+        .attach_printable("Could not insert data type")
+    }
+}
+
 #[async_trait]
 impl Datastore for PostgresDatabase {
-    async fn create_data_type() -> Result<(), DatastoreError> {
-        todo!()
+    async fn create_data_type(
+        &self,
+        data_type: DataType,
+        created_by: AccountId,
+    ) -> Result<Qualified<DataType>, DatastoreError> {
+        let id = Identifier {
+            base_id: BaseId::new(Uuid::new_v4()),
+            version_id: VersionId::new(Uuid::new_v4()),
+        };
+
+        self.insert_version_id(&id).await?;
+
+        self.insert_data_type(&id, &data_type, created_by).await?;
+
+        Ok(Qualified {
+            id,
+            inner: data_type,
+            created_by,
+        })
     }
 
-    async fn get_data_type() -> Result<(), DatastoreError> {
-        todo!()
+    async fn get_data_type(&self, id: Identifier) -> Result<Qualified<DataType>, DatastoreError> {
+        let DataTypeRow {
+            schema: data_type,
+            created_by,
+            ..
+        } = sqlx::query_as(
+            r#"
```

Comment:
I'm not sure how I feel about writing our SQL inline in strings in functions. In the long run it'd be good to see if we can get these version controlled in actual `sql` files or something so we can benefit from more SQL tooling like linting, intellisense, etc.

Reply:
Just for completeness, sqlx does have a macro that imports from sql files, [`sqlx::query_file_as!`](https://docs.rs/sqlx/latest/sqlx/macro.query_file_as.html). Those do compile-time checks on the query AFAIK.